A32NX/html_ui/Pages/A32NX_Core/A32NX_TransitionManager.js

```diff
@@ -0,0 +1,88 @@
+class A32NX_TransitionManager {
+    init() {
+        this.UpdateTimer = 0;
+        console.log('A32NX_TransitionManager init');
+        const mode = NXDataStore.get("CONFIG_TRANSALT", "AUTO");
+        if (mode !== "AUTO") {
+            this.transitionManual();
+        }
+        if (mode === "AUTO") {
+            this.UpdateTimer = 130;
+        }
+    }
+    update(_deltaTime, _core) {
+        this.transitionSelector();
+    }
+    transitionSelector(_deltaTime) {
+        this.UpdateTimer += _deltaTime / 1000;
+        const mode = NXDataStore.get("CONFIG_TRANSALT", "AUTO");
+        if (mode === "AUTO") {
+            const departureICAO = NXDataStore.get("PLAN_ORIGIN", "");
+            const arrivalICAO = NXDataStore.get("PLAN_DESTINATION", "");
+            if (this.UpdateTimer > 120) {
+                this.departureLogic(departureICAO);
+                this.arrivalLogic(arrivalICAO);
+                //this.transLVL();
+                this.UpdateTimer = 0;
```

Comment:
why do you need to check (have/use an update routine?)
What about calling the update/change when changing departure/arrival?

Reply:
sure! i fix it today :)